credential issuance: clarification about user auth at authz endpoint and idp hint (#1187)
italia/eidas-it-wallet-docs: docs/en/credential-issuance-l2plus.rst

Authorization Details
"""""""""""""""""""""

> The JWT Request Object MUST contain the same parameters as defined in :ref:`credential-issuance-endpoint:Pushed Authorization Request Endpoint`. When the User requests a PID using eID Substantial Authentication with MRTD Verification, the Wallet Instance MAY include an additional **Authorization Details Object** in the ``authorization_details`` parameter, with the structure and claims as defined in the Table of the JWT Request parameters of Section :ref:`credential-issuance-endpoint:Pushed Authorization Request Endpoint`. Such object constitutes an optional hint for the Authorization Server, according to :ref:`credential-issuance-endpoint:User Authentication Method Selection`.

Reviewer: ```suggestion
The JWT Request Object MUST contain the same parameters as defined in :ref:`credential-issuance-endpoint:Pushed Authorization Request Endpoint`. When the User requests a PID using eID Substantial Authentication with MRTD Verification, the Wallet Instance SHOULD include an additional **Authorization Details Object** in the ``authorization_details`` parameter, with the structure and claims as defined in the Table of the JWT Request parameters of Section :ref:`credential-issuance-endpoint:Pushed Authorization Request Endpoint`. Such object constitutes an optional hint for the Authorization Server, according to :ref:`credential-issuance-endpoint:User Authentication Method Selection`.
```

If the Wallet has presented a discovery page and the User has made a decision, then the Wallet must include an additional **Authorization Details Object** in the ``authorization_details`` parameter

Author: the optionality would be better explained using a MAY

When the ``it_l2+document_proof`` object is not present in the ``authorization_details`` array, or when the Authorization Server does not support or does not accept the related hint, the PID Provider MUST authenticate the User according to its authentication policies, as defined in :ref:`credential-issuance-endpoint:User Authentication Method Selection`. In the absence of a User selection via a Credential Issuer discovery page, the default authentication method MUST be CIEid LoA High.
The PAR Response and the Authorization Request are the same as in the IT-Wallet Specification.

Phase 2: Primary Authentication
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Upon successful processing of the PAR, the Authorization Server redirects the User Agent to the 1Identity Provider for primary authentication. If the request contains a supported and accepted ``idphinting``, the Authorization Server SHOULD use it; otherwise it MUST use a Identity Provider that it has configured or selected via its own discovery page. The User completes the authentication flow and the Authorization Server correlates the authenticated identity with the active OAuth session.

Author: ```suggestion
Upon successful processing of the PAR, the Authorization Server redirects the User Agent to the Identity Provider for primary authentication. If the request contains a supported and accepted ``idphinting``, the Authorization Server SHOULD use it; otherwise it MUST use a Identity Provider that it has configured or selected via its own discovery page. The User completes the authentication flow and the Authorization Server correlates the authenticated identity with the active OAuth session.
```

    client_id=47b982369791d08003a7283f059cb0d1&
    request=eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJqdGkiOiJmODU1NWNlYi1jNjVjLTQwMjUtOTM3OC1iNjY3MmI2MTQ5YWYiLCJhdWQiOiJodHRwczovL3BpZC1wcm92aWRlci5leGFtcGxlLm9yZyIsImlhdCI6MTcxNTg0MjU2MCwiZXhwIjoxNzE1ODQyODYwLCJyZXNwb25zZV90eXBlIjoiY29kZSIsInJlc3BvbnNlX21vZGUiOiJmb3JtX3Bvc3Quand0IiwiY2xpZW50X2lkIjoiNDdiOTgyMzY5NzkxZDA4MDAzYTcyODNmMDU5Y2IwZDEiLCJpc3MiOiI0N2I5ODIzNjk3OTFkMDgwMDNhNzI4M2YwNTljYjBkMSIsInN0YXRlIjoiZnlaaU9MOUxmMkNlS3VOVDJKenhpTFJEaW5rMHVQY2QiLCJjb2RlX2NoYWxsZW5nZSI6IkU5TWVsaG9hMk93dkZyRU1USmd1Q0hhb2VLMXQ4VVJXYnVHSlNzdHctY00iLCJjb2RlX2NoYWxsZW5nZV9tZXRob2QiOiJTMjU2Iiwic2NvcGUiOiJwaWQiLCJhdXRob3JpemF0aW9uX2RldGFpbHMiOlt7InR5cGUiOiJvcGVuaWRfY3JlZGVudGlhbCIsImNyZWRlbnRpYWxfY29uZmlndXJhdGlvbl9pZCI6ImRjX3NkX2p3dF9waWQifSx7InR5cGUiOiJpdF9sMitkb2N1bWVudF9wcm9vZiIsIm11bHRpX3N0ZXBfbWV0aG9kIjoibXJ0ZCtpYXMiLCJpZHBoaW50aW5nIjoiaHR0cHM6Ly9pZHAuZXhhbXBsZS5vcmciLCJtdWx0aV9zdGVwX3JlZGlyZWN0X3VyaSI6Imh0dHBzOi8vc3RhcnQud2FsbGV0LmV4YW1wbGUub3JnL2NoYWxsZW5nZSJ9XSwicmVkaXJlY3RfdXJpIjoiaHR0cHM6Ly9zdGFydC53YWxsZXQuZXhhbXBsZS5vcmcifQ.AuthRequestSign456_NoKidJWTSignature-abc123def456ghi789jkl012mno345pqr678stu901vwx234yz567

> When the ``it_l2+document_proof`` object is not present in the ``authorization_details`` array, or when the Authorization Server does not support or does not accept the related hint, the PID Provider MUST authenticate the User according to its authentication policies, as defined in :ref:`credential-issuance-endpoint:User Authentication Method Selection`. In the absence of a User selection via a Credential Issuer discovery page, the default authentication method MUST be CIEid LoA High.

Reviewer: ```suggestion
When the ``it_l2+document_proof`` object is not present in the ``authorization_details`` array, or when the Authorization Server does not support or does not accept the related hint, the PID Provider MUST authenticate the User according to its authentication policies, as defined in :ref:`credential-issuance-endpoint:User Authentication Method Selection`. In the absence of a User selection via a Credential Issuer discovery page, the process cannot continue.
```
As I said in a previous comment, if the user does not make a selection, no authentication can take place. There is no default authentication method that can be applied. The process simply stops.

Author: No as my previous comment

The PAR Response and the Authorization Request are the same as in the IT-Wallet Specification.

Phase 2: Primary Authentication
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Upon successful processing of the PAR, the Authorization Server redirects the User Agent to the 1Identity Provider for primary authentication. If the request contains a supported and accepted ``idphinting``, the Authorization Server SHOULD use it; otherwise it MUST use a Identity Provider that it has configured or selected via its own discovery page. The User completes the authentication flow and the Authorization Server correlates the authenticated identity with the active OAuth session.

Reviewer: ```suggestion
Upon successful processing of the PAR, the Authorization Server redirects the User Agent to the 1Identity Provider for primary authentication. If the request contains a supported and accepted ``idphinting``, the Authorization Server MUST use it; otherwise it MUST use a Identity Provider that it has configured or selected via its own discovery page. The User completes the authentication flow and the Authorization Server correlates the authenticated identity with the active OAuth session.
```
Under what circumstances might the AS choose not to follow the hint if supported and accepted? It is important to not. It is important to ensure that the User does not have to make the same choice on two consecutive discovery pages.

Author: rejected as my preivous comment

the idp hinting is optional for both the wallet solution and the AS
